chromadb/segment/impl/vector/local_persistent_hnsw.py

```diff
@@ -251,7 +253,11 @@ def _write_records(self, records: Sequence[EmbeddingRecord]) -> None:
                     self._brute_force_index.clear()
 
     @override
-    def count(self) -> int:
+    def count(
+        self,
+        where: Optional[Where] = {},
```

Comment:
Hmmm - the fact that these segment implementations just ignore these things feels wrong. It's odd that its not respected. We should probably override count in the MetadataReader so that only Metadata supports filtered count().

Reply:
It felt a bit odd while doing it, kept it to get a review on it. Should I override the count in MetadataReader or define a different count in VectorReader?

Reply:
I prefer overriding it in metadata reader - wdyt?

Reply:
Yeah, it looks more elegant. Will go with it.